content/blog/2023-12-01-task-lifecycle.markdown

```diff
@@ -0,0 +1,470 @@
+Title: Lifecycle of a kernel task
+Date: 2023-12-01 08:00:00
+Authors: Benjamin Gray
+Category: Development
+Tags: linux
+
+## Introduction
+
+CPU cores are very limited in number. Right now my computer tells me it's
+running around 500 processes, and I definitely do not have that many CPU cores.
+The ability for the operating system to virtualise the concept of an 'execution
+unit' and swap them in and out of running on the limited pool CPU cores is one
+of the foundations of modern computing.
```

Comment:
~foundations of modern computing~ essential requirements of an operating system.
Don't have to take this one if you prefer the former.

Reply:
I like foundation, it's difficult to imagine modern computing (whatever that means) without this functionality. You could argue there are embedded systems that might not have tasks, but eh